src/main/java/com/zufar/onlinestore/product/endpoint/ProductsEndpoint.java

```diff
@@ -1,64 +1,59 @@
 package com.zufar.onlinestore.product.endpoint;
 
-import com.zufar.onlinestore.product.converter.ProductInfoDtoConverter;
-import com.zufar.onlinestore.product.dto.ProductInfoDto;
-import com.zufar.onlinestore.product.entity.ProductInfo;
-import com.zufar.onlinestore.product.repository.ProductInfoRepository;
-
+import com.zufar.onlinestore.product.api.ProductApi;
+import com.zufar.onlinestore.product.dto.ProductPaginationDto;
+import com.zufar.onlinestore.product.dto.ProductResponseDto;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.ResponseEntity;
 import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
-import java.util.Collection;
-import java.util.Optional;
-
-import lombok.RequiredArgsConstructor;
-import lombok.extern.slf4j.Slf4j;
+import java.util.UUID;
 
 @Slf4j
 @RestController
 @RequiredArgsConstructor
 @Validated
 @RequestMapping(value = "/api/products")
 public class ProductsEndpoint {
-    private final ProductInfoRepository productInfoRepository;
-    private final ProductInfoDtoConverter productInfoDtoConverter;
+
+    private final ProductApi productInfoService;
 
     @GetMapping("/{id}")
-    public ResponseEntity<ProductInfoDto> getProductInfoById(@PathVariable("id") final String id) {
-        log.info("Received request to get the ProductInfo with id - {}.", id);
-        Optional<ProductInfo> ProductInfo = productInfoRepository.findById(Integer.parseInt(id));
-        if (ProductInfo.isEmpty()) {
-            log.info("the ProductInfo with id - {} is absent.", id);
-            return ResponseEntity.notFound()
-                    .build();
-        }
-        ProductInfoDto ProductInfoDto = productInfoDtoConverter.convertToDto(ProductInfo.get());
-        log.info("the ProductInfo with id - {} was retrieved - {}.", id, ProductInfoDto);
+    public ResponseEntity<ProductResponseDto> getProductById(@PathVariable("id") final String id) {
+        log.info("Received request to get the Product with id - {}.", id);
+        UUID convertedId = UUID.fromString(id);
         return ResponseEntity.ok()
-                .body(ProductInfoDto);
+                .body(productInfoService.getProduct(convertedId));
```

Comment:
Create separate variable for that 
like that
```
Product product = productInfoService.getProduct(convertedId);
return ResponseEntity.ok()
                 .body(product);
```

Reply:
Fixed.